File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/ui/base/activity/ConnectivityChangeReceiver.java
package com.lattis.ellipse.presentation.ui.base.activity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

/**
 * Created by ssd3 on 10/6/17.
 */

public class ConnectivityChangeReceiver extends BroadcastReceiver {

    private OnConnectivityChangedListener listener;

    public ConnectivityChangeReceiver(OnConnectivityChangedListener listener) {
        this.listener = listener;
    }

    @Override
    public void onReceive(Context context, Intent intent) {

        if(isNetworkAvailable(context)) {
            listener.onConnectivityChanged(true);
        }else{
            listener.onConnectivityChanged(false);
        }

    }

    public interface OnConnectivityChangedListener {
        void onConnectivityChanged(boolean isConnected);
    }



    private boolean isNetworkAvailable(Context context) {
        try {
            ConnectivityManager cm =
                    (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

            NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
            boolean isConnected = activeNetwork != null &&
                    activeNetwork.isConnectedOrConnecting();
            return isConnected;
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
        return false;
    }

}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/presentation/base/activity/ConnectivityChangeReceiver.kt
package com.lattis.lattis.presentation.base.activity

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager

class ConnectivityChangeReceiver(private val listener: OnConnectivityChangedListener) :
    BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (isNetworkAvailable(context)) {
            listener.onConnectivityChanged(true)
        } else {
            listener.onConnectivityChanged(false)
        }
    }

    interface OnConnectivityChangedListener {
        fun onConnectivityChanged(isConnected: Boolean)
    }

    private fun isNetworkAvailable(context: Context): Boolean {
        try {
            val cm =
                context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val activeNetwork = cm.activeNetworkInfo
            return activeNetwork != null &&
                    activeNetwork.isConnectedOrConnecting
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return false
    }

}